src/helpers/util.js

```diff
@@ -39,5 +39,40 @@ module.exports = {
       ret.push(row)
     }
     return ret
+  },
+
+  /*
+    Binary Search:
+    Assumes that @arg array is sorted increasingly
+    returns the smallest i such that target >= changes[i]
+    returns arary.length - 1 (if all elements in array are smaller than target)
```

Comment:
I think it should return `array.length`, otherwise this conditions conflicts with the previous one.